### PR TITLE
Code review: Project 2: Members Only!

### PR DESCRIPTION
**Pull request by bafiam**
#### What does this PR do?
1. Consist of the member only final code
- users can login 
- create a post
- view post without login
- view post author when logged in
#### How should this be manually tested?
- Clone the repo then navigate to the project directory, follow the instruction on the readme
- edit the seed file in the db folder to add a user detail of your choice
- To add a new user to the database do a seed command using `rails db:seed`

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user&.authenticate(params[:session][:password])
+      session[:user_id] = user.id
```
Comment (maelfosso) [inline app/controllers/sessions_controller.rb]: Forget about using `session`, please.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,38 @@
...
+
+  def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      session[:user_id] = user.id
+      @current_user = user
+    end
+  end
```
Comment (maelfosso) [inline app/controllers/application_controller.rb]: Too long. Rewrite it by keeping in my that when a user is connected only the remember token is set up in cookies.permanent.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,2 +1,38 @@
...
+
+  def forget_session(user)
+    user.delete_session_token
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+  end
+
+  def logout
+    forget_session(current_user)
+    session.delete(:user_id)
```
Comment (maelfosso) [inline app/controllers/application_controller.rb]: No needed

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,
Thank you for the simplicity of your work.
To move forward, you need some changes to do

1. Please, forget about using `session` in this project. Only think about using `cookies.permanent`. 
The `remember_me token` the project describes consists of storing a hash in a cookie and using it to determine in a user is logged in. Do not confuse it with the Remember me \[ \] checkbox that many websites show. That is a more complex solution, which is **not required** here.

2. You forgot to define the method to set user `#current_user=(user)`  as indicated in that [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/current_user_requirement.png)

3. In your remember token, you forgot to set the current user, to update the current user with his remember token.

4. In the `http://localhost:3000`, `Sign up now!` is not working, no link there! I was obliged to create a user through `rails console`

5. Read the comment below.

Kindly make the suggested changes and submit a new code review request.
Thanks

Mael FOSSO

**Comment by simoxsimo** [comment]
## Problems are fixed !!

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

You can merge it :+1: 
Good luck with the next :handshake: 

Mael FOSSO

**Pull request merged**